src/porepy/applications/convergence_analysis.py

```diff
@@ -0,0 +1,439 @@
+                setattr(setup.results[-1], "num_dofs", setup.equation_system.num_dofs())
+                setattr(setup.results[-1], "cell_diameter", setup.mdg.diameter())
+            else:
+                # Run time-dependent model
+                pp.run_time_dependent_model(setup, deepcopy(self.model_params[level]))
+                # Complement information in results
+                setattr(setup.results[-1], "num_dofs", setup.equation_system.num_dofs())
+                setattr(setup.results[-1], "cell_diameter", setup.mdg.diameter())
+                setattr(setup.results[-1], "dt", setup.time_manager.dt)
+
+            convergence_results.append(setup.results[-1])
+        return convergence_results
+
+    def export_errors_to_txt(
+        self,
+        list_of_results: list,
+        variables_to_export: Optional[list[str]] = None,
+        file_name="error_analysis.txt",
+    ) -> None:
+        """Write errors into a ``txt`` file.
+
+        The format is the following one:
+
+            - First column contains the cell diameters.
+            - Second column contains the time steps (if the model is time-dependent).
+            - The rest of the columns contain the errors for each variable in
+              ``variables``.
+
+        Parameters:
+            list_of_results: List containing the results of the convergence analysis.
+                Typically, the output of :meth:`~run_analysis`.
+            variables_to_export: names of the variables for which the TXT file will be
+                generated. If ``variables`` is not given, all the variables present
+                in the txt file will be collected.
+            file_name: Name of the output file. Default is "error_analysis.txt".
+
+        """
+        # Filter variables from the list of results
+        var_names: list[str] = self._filter_variables_from_list_of_results(
+            list_of_results=list_of_results,
+            variables=variables_to_export,
+        )
+
+        # Filter errors to be exported
+        errors_to_export: dict[str, np.ndarray] = {}
+        for name in var_names:
+            # Loop over lists of results
+            var_error: list[float] = []
+            for result in list_of_results:
+                var_error.append(getattr(result, name))
+            # Append to the dictionary
+            errors_to_export[name] = np.array(var_error)
+
+        # Prepare to export
+        list_of_txt_data: list[TxtData] = []
+        # Append cell diameters
+        cell_diameters = np.array([result.cell_diameter for result in list_of_results])
+        list_of_txt_data.append(TxtData(header="cell_diameter", array=cell_diameters))
+        # Append time steps (if the analysis employs a time-dependent model)
+        if self._is_time_dependent:
+            time_steps = np.array([result.dt for result in list_of_results])
+            list_of_txt_data.append(TxtData(header="time_step", array=time_steps))
```

Comment:
This uses the default number format defined in TxtData, right? What about defining a helper method say 
`_data_format(header) -> str:
   return TxtData.format
`
Then, if I have special needs for some quantity, I inherit from this class and override the helper method to return something else for that quantity.